src/x/server/server_test.go

```diff
@@ -39,14 +40,21 @@ const (
 )
 
 // nolint: unparam
-func testServer(addr string) (*server, *mockHandler, *int32, *int32) {
+func testServer(addr string, tlsMode TLSMode, mTLSEnabled bool) (*server, *mockHandler, *int32, *int32) {
```

Comment:
Nit: could factor out a small `testTLSOptions` struct here; save some typing for non TLS tests.

Reply:
Split it into two functions: one for plain TCP server and another for TLS server.